app/controllers/posts_controller.rb

```diff
@@ -27,6 +27,17 @@ def create
     end
   end
 
+  def remove_post
+    @post = Post.find(params[:id])
+    @post.destroy
+    @user = current_user
+    @user.save
+
+    respond_to do |format|
+      format.html { redirect_to(user_posts_url) }
+    end
+  end
+
```

Comment:
- [ ] you should use CanCanCan for Post authorization in the post_controller by adding: ` load_and_authorize_resource`

Reply:
Thanks